addon_service/addon_operation_invocation/serializers.py

```diff
@@ -66,22 +85,42 @@ class Meta:
     )
 
     included_serializers = {
+        "thru_account": "addon_service.serializers.AuthorizedStorageAccountSerializer",
         "thru_addon": "addon_service.serializers.ConfiguredStorageAddonSerializer",
         "operation": "addon_service.serializers.AddonOperationSerializer",
         "by_user": "addon_service.serializers.UserReferenceSerializer",
     }
 
     def create(self, validated_data):
-        _thru_addon: ConfiguredStorageAddon = validated_data["thru_addon"]
+        _thru_addon = validated_data.get("thru_addon")
+        _thru_account = validated_data.get("thru_account")
+        if _thru_addon is None and _thru_account is None:
+            raise ValidationError("must include either 'thru_addon' or 'thru_account'")
+        if (
+            _thru_addon
+            and _thru_account
+            and (_thru_addon.base_account != _thru_account)
+        ):
+            raise ValidationError("mismatched 'thru_addon' and 'thru_account'")
```

Comment:
This error is also raised in validation at the model level. Worth checking in both places? Or cut it here?

Reply:
alternatively, only allow one or the other? I may get too much mileage out of `bool(x) == bool(y)`, though...